src/ColorPickerBubbleMenu/index.tsx

```diff
@@ -0,0 +1,145 @@
+
+  if (!editor?.isEditable) {
+    return null;
+  }
+
+  if (!("colorPickerBubbleMenuHandler" in editor.storage)) {
+    throw new Error(
+      "You must add the ColorPickerBubbleMenuHandler extension to the useEditor `extensions` array in order to use this component!"
+    );
+  }
+  const handlerStorage = editor.storage
+    .colorPickerBubbleMenuHandler as ColorPickerBubbleMenuHandlerStorage;
+
+  // Update the menu step if the bubble menu state has changed
+  const menuState = handlerStorage.state;
+
+  // Determine if all of the selected content shares the same set color.
+  const allCurrentTextStyleAttrs: TextStyleAttrs[] =
+    getAttributesForEachSelected(editor.state, "textStyle");
+  const isTextStyleAppliedToEntireSelection = !!editor.isActive("textStyle");
+  const currentColors: string[] = allCurrentTextStyleAttrs.map(
+    // Treat any null/missing color as the default color
+    // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+    (attrs) => attrs.color || defaultTextColor
+  );
+  if (!isTextStyleAppliedToEntireSelection) {
+    // If there is some selected content that does not have textStyle, we can
+    // treat it the same as a selected textStyle mark with color set to the
+    // default
+    currentColors.push(defaultTextColor);
+  }
+  const numUniqueCurrentColors = new Set(currentColors).size;
+
+  let currentColor;
+  if (numUniqueCurrentColors === 1) {
+    // There's exactly one color in the selected content, so show that
+    currentColor = currentColors[0];
+  } else if (numUniqueCurrentColors > 1) {
+    // There are multiple colors (either explicitly, or because some of the
+    // selection has a color set and some does not and is using the default
+    // color). Similar to other rich text editors like Google Docs, we'll treat
+    // this as "unset" and not show a color indicator in the button or a
+    // "current" color when interacting with the color picker.
+    currentColor = "";
+  } else {
+    // Since no color was set anywhere in the selected content, we should show
+    // the default color
+    currentColor = defaultTextColor;
+  }
+
+  return (
+    <ControlledBubbleMenu
+      editor={editor}
+      open={menuState}
+      {...handlerStorage.bubbleMenuOptions}
+      {...controlledBubbleMenuProps}
+    >
+      <ColorPickerPopperBody
+        value={currentColor}
+        onSave={(newColor: string) => {
+          if (handlerStorage.bubbleMenuOptions?.mode === ColorPickerMode.Text) {
+            editor.chain().focus().setColor(newColor).run();
+          } else {
+            // Highlight
+            if (newColor) {
+              editor.chain().focus().setHighlight({ color: newColor }).run();
+            } else {
+              editor.chain().focus().unsetHighlight().run();
+            }
+          }
+          editor.commands.closeColorPickerBubbleMenu();
+        }}
+        onCancel={editor.commands.closeColorPickerBubbleMenu}
```

Comment:
The behavior of closing the new bubble menu seems unintuitive to me and doesn't match the original behavior of the color picker menu. Clicking outside the picker doesn't close/cancel it, so the user has to click "Cancel" before it will go away. Maybe that could be useful functionality in some situations, but it doesn't seem like it should be the default. It seems slightly more confusing because the picker now appears near the selected editor content, rather than near the menu button, but doesn't move even as your selection changes (despite staying open).